iset.mm

```diff
@@ -148169,6 +148242,320 @@ mathematician Ptolemy (Claudius Ptolemaeus).  This particular version is
   $}
 
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Logarithms to an arbitrary base
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+
+  Define "log using an arbitrary base" function and then prove some of its
+  properties.  As with ~ df-relog this is for real logarithms rather than
+  complex logarithms.
+
+  Metamath doesn't care what letters are used to represent classes.  Usually
+  classes begin with the letter "A", but here we use "B" and "X"
+  to more clearly distinguish between "base" and "other parameter of log".
+
+  There are different ways this could be defined in Metamath.  The approach
+  used here is intentionally similar to existing 2-parameter Metamath functions
+  (operations): ` ( B logb X ) `  where ` B ` is the base and ` X ` is the
+  argument of the logarithm function.  An alternative would be to support the
+  notational form ` ( ( logb `` B ) `` X ) `; that looks a little more like
+  traditional notation.
+
+$)
+
+  $c logb $. $( Logarithm generalized to an arbitrary base. $)
+
+  $( Extend class notation to include the logarithm generalized to an arbitrary
+     base. $)
+  clogb $a class logb $.
+
+  ${
+    $d x y $.
+    $( Define the ` logb ` operator.  This is the logarithm generalized to an
+       arbitrary base.  It can be used as ` ( B logb X ) ` for "log base B of
+       X".  In the most common traditional notation, base B is a subscript of
+       "log".  The definition will only be useful where ` x ` is a positive
+       real apart from one and where ` y ` is a positive real, so the choice of
+       ` ( CC \ { 0 , 1 } ) ` and ` ( CC \ { 0 } ) ` is somewhat arbitrary (we
+       adopt the definition used in set.mm).  (Contributed by David A. Wheeler,
+       21-Jan-2017.) $)
+    df-logb $a |- logb = ( x e. ( CC \ { 0 , 1 } ) , y e. ( CC \ { 0 } ) |->
```

Comment:
to be consistent, `x` should be replaced by `b`, and `y` by `x `.

Reply:
I wouldn't object if someone wants to do that (in set.mm and iset.mm, presumably), but there's definitely precedent for using x and y in definitions even if we use other letters elsewhere.